cmd/jaeger/internal/integration/span_writer.go

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package integration
+
+import (
+	"context"
+
+	"github.com/jaegertracing/jaeger/model"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+	jaeger2otlp "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/translator/jaeger"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/testbed/testbed"
+)
+
+var _ spanstore.Writer = (*spanWriter)(nil)
+
+// SpanWriter utilizes the OTLP exporter to send span data to the Jaeger-v2 receiver
+type spanWriter struct {
+	testbed.TraceDataSender
```

Comment:
do you want spanWriter __to be__ a TraceDataSender (i.e. expose all of the sender's API), or is it just an implementation detail of the spanWriter? If it's the latter, you don't want to embed the struct, but have a named field.

Reply:
I see. It is just an implementation detail actually, named it in the recent changes.